src/DurableTask.AzureStorage/Partitioning/TablePartitionManager.cs

```diff
@@ -0,0 +1,788 @@
+                    {
+                        claimedLease = TryClaimLease(partition);
+                        CheckOtherWorkerLease(partition, partitionDistribution, response);
+                        await CheckOwnershipLease(partition, response);
+                    }
+                    else// If shutdown is requested, stop claiming and balancing partitions. Instead, drain and release all ownership partitions.
+                    {
+                        await TryDrainAndReleaseAllPartitions(partition, response);
+                    }
+
+                    // Update the table if the lease is claimed, stolen, renewed, drained or released.
+                    if (claimedLease || stoleLease || renewedLease || drainedLease || releasedLease)
+                    {
+                        try
+                        {
+                            await this.partitionTable.UpdateEntityAsync(partition, etag, (TableUpdateMode)1);
+
+                            if (claimedLease)
+                            {
+                                await this.service.OnTableLeaseAcquiredAsync(partition);
+                            }
+
+                            this.LogHelper(partition, claimedLease, stoleLease, renewedLease, drainedLease, releasedLease, previousOwner);
+                        }
+                        // Exception for failed table update due to ETag expired. 
+                        // Re-read the table immediately to get a new ETag.
+                        catch (RequestFailedException ex) when (ex.Status == 412)
+                        {
+                            this.settings.Logger.PartitionManagerInfo(
+                                this.storageAccountName,
+                                this.settings.TaskHubName,
+                                this.settings.WorkerId,
+                                partition.RowKey,
+                                $"Failed to update table entry due to an Etag mismatch. Failed ETag value: '{etag}'.");
+                            throw;
+                        }
+                        // Eat any exceptions
+                        catch (Exception exception)
+                        {
+                            this.settings.Logger.PartitionManagerError(
+                                this.storageAccountName,
+                                this.settings.TaskHubName,
+                                this.settings.WorkerId,
+                                partition.RowKey,
+                                $"Unexpected error occurred when updating the table: {exception.Message}");
+                        }
+                    }
+
+                    // Check if a lease is available for the current worker.
+                    bool TryClaimLease(TableLease partition)
+                    {
+                        //Check if the partition is empty, expired or stolen by the current worker and claim it.
```

Comment:
nit: let's add a space between the "//" and the first character.
Also added a minor phrasing suggestion
```suggestion
                        // Check if the partition is empty, expired, or stolen by the current worker. If any of these apply, claim it.
```